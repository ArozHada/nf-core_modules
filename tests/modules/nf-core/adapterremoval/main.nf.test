nextflow_process {

    name "Test Process ADAPTERREMOVAL"
    script "modules/nf-core/adapterremoval/main.nf"
    config "tests/modules/nf-core/adapterremoval/nextflow.config"
    process "ADAPTERREMOVAL"
    tag "adapterremoval"

    test("Single End") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:true, collapse:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                            ]

                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("single_end") }
            )
        }

    }

    test("Paired End") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false, collapse:false ], // meta map
                           [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                               file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
                           ]

                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("paired_end") }
            )
        }

    }

    test("Paired End Collapse") {

        options "--collapse"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                            [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                                file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
                            ]

                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("paired_end_collapse") }
            )
        }

    }

    test("Paired End Collapse With AdapterList") {

        options "--collapse"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false, collapse:false ], // meta map
                            [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                                file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
                            ]
                input[1] = file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/adapterremoval/adapterremoval_adapterlist.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("paired_end_collapse_adapterlist") }
            )
        }

    }

}
