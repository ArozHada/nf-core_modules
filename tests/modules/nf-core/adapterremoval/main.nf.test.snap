{
    "paired_end_collapse_adapterlist": {
        "content": [
            [
                "test.discarded.fastq.gz:md5,7029066c27ac6f5ef18d660d5741979a",
                "test.pair1.truncated.fastq.gz:md5,294a6277f0139bd597e57c6fa31f39c7",
                "test.pair2.truncated.fastq.gz:md5,de7b38e2c881bced8671acb1ab452d78",
                "test.settings:md5,36d47d9b40dbc178167d1ae0274d18f3",
                "versions.yml:md5,00bcc9f0b864b96eeee21bc11773ee67"
            ]
        ],
        "timestamp": "2023-04-05T17:39:56+0000"
    },
    "single_end": {
        "content": [
            [
                "test.discarded.fastq.gz:md5,7029066c27ac6f5ef18d660d5741979a",
                "test.settings:md5,2fd3d5d703b63ba33a83021fccf25f77",
                "test.truncated.fastq.gz:md5,62139afee94defad5b83bdd0b8475a1f",
                "versions.yml:md5,00bcc9f0b864b96eeee21bc11773ee67"
            ]
        ],
        "timestamp": "2023-04-05T17:39:56+0000"
    },
    "paired_end": {
        "content": [
            [
                "test.discarded.fastq.gz:md5,7029066c27ac6f5ef18d660d5741979a",
                "test.pair1.truncated.fastq.gz:md5,294a6277f0139bd597e57c6fa31f39c7",
                "test.pair2.truncated.fastq.gz:md5,de7b38e2c881bced8671acb1ab452d78",
                "test.settings:md5,b8a451d3981b327f3fdb44f40ba2d6d1",
                "versions.yml:md5,00bcc9f0b864b96eeee21bc11773ee67"
            ]
        ],
        "timestamp": "2023-04-05T17:39:56+0000"
    },
    "paired_end_collapse": {
        "content": [
            [
                "test.discarded.fastq.gz:md5,7029066c27ac6f5ef18d660d5741979a",
                "test.pair1.truncated.fastq.gz:md5,294a6277f0139bd597e57c6fa31f39c7",
                "test.pair2.truncated.fastq.gz:md5,de7b38e2c881bced8671acb1ab452d78",
                "test.settings:md5,b8a451d3981b327f3fdb44f40ba2d6d1",
                "versions.yml:md5,00bcc9f0b864b96eeee21bc11773ee67"
            ]
        ],
        "timestamp": "2023-04-05T17:39:56+0000"
    }
}